### PR TITLE
Output Single iOS Framework Distribution

### PR DESCRIPTION
**Pull request by joeydong**
This switches the strategy for handling iOS application extension hosts to using a runtime check instead of providing a separate framework output. See commit messages for details.

We still have a bunch of validation to do before this is suitable for merging but I'm opening the PR for visibility.

By the way, have we considered using a mocking framework to simplify the unit test writing?

### CODE REVIEW

File: ADAL/src/ui/ios/UIAlertView+Additions.m [hunk 1]
```diff
@@ -1,51 +1,63 @@
...
+            bundle = [NSBundle mainBundle];
+        }
+
+        alert = [[UIAlertView alloc] initWithFrame:CGRectZero];
+        alert.title = NSLocalizedStringFromTableInBundle(@"Enter your credentials", nil, bundle, nil);
+        alert.alertViewStyle = UIAlertViewStyleLoginAndPasswordInput;
+        
+        [alert addButtonWithTitle:NSLocalizedStringFromTableInBundle(@"Cancel", nil, bundle, nil)];
+        [alert addButtonWithTitle:NSLocalizedStringFromTableInBundle(@"Login", nil, bundle, nil)];
+        [alert setCancelButtonIndex:0];
```
Comment (joeydong) [inline ADAL/src/ui/ios/UIAlertView+Additions.m]: I'm not totally confident about this change. How do we reproduce this flow?

**Comment by joeydong** [comment]
I'm not sure how the frameworks are getting imported in the sample project. How do I fix that?

File: ADAL/src/utils/ios/ADAppExtensionUtil.m [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+@implementation ADAppExtensionUtil
+
++ (BOOL)isExecutingInAppExtension
+{
+    NSString* mainBundlePath = [[NSBundle mainBundle] bundlePath];
+
+    if (mainBundlePath.length == 0)
+    {
+        AD_LOG_ERROR(@"Expected `[[NSBundle mainBundle] bundlePath]` to be non-nil. Defaulting to non-application-extension safe API.", AD_ERROR_UNEXPECTED, nil, nil);
```
Comment (eddiekim) [inline ADAL/src/utils/ios/ADAppExtensionUtil.m]: Not sure that this should be an error since this path will be part of the normal flow from an extension.

Comment (joeydong) [inline ADAL/src/utils/ios/ADAppExtensionUtil.m]: The `[NSBundle mainBundle]` returns something in both cases:

- When the host is an app, the `bundlePath` is something like `file://.../app-ios.app`
- When the host is an extension, the `bundlePath` is something like `file://.../today-widget.appex`

File: ADAL/src/ui/ios/UIAlertView+Additions.m (hunk 1 above)
Comment (RPangrle) [inline ADAL/src/ui/ios/UIAlertView+Additions.m]: You need to get a server to send you an NTLM challenge. The only way I have to do it involves using a MS STS while on corpnet and your corpnet account. I'll try to repro and send you directions.

**Comment by RPangrle** [comment]
@joeydong, the SampleSwiftApp scheme itself is set to build the ADAL framework first and then the Sample Swift App. Considering that error doesn't repro cleanly I'm wondering if xcodebuild is getting confused from some other state laying around.

File: ADAL/src/utils/ios/ADAppExtensionUtil.m (hunk 2 above)
Comment (eddiekim) [inline ADAL/src/utils/ios/ADAppExtensionUtil.m]: oh sorry, I misunderstood. never mind

**Comment by joeydong** [comment]
thx for reviewing @RPangrle / @eddiekim

**Pull request merged**